instagram/src/App.js

```diff
@@ -0,0 +1,26 @@
+import React, { Component } from "react";
+import logo from "./logo.svg";
+import "./App.css";
+import PostsPage from "./components/PostsPage/PostsPage.js";
+import Authenticate from "./components/Authentication/Authenticate";
+import LoginPage from "./components/LoginPage/Login";
+
+class App extends Component {
+  constructor() {
+    super();
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <PostsPage />
+      </div>
+    );
+  }
+}
+
+const HOCApp = Authenticate(App);
+
+return <HOCApp />;
+
+export default App;
```

Comment:
For lines 22-26 you're on the right track, however you don't need lines 22-24. In your export you could just say `export default Authenticate(App)`

Reply:
what would the Authenticate(App) do in this case? does it export the App component by executing it through Authenticate first every time ??